### PR TITLE
Exasol data types

### LINKED ISSUE #6: Add support for all Exasol data types
See https://docs.exasol.com/sql_references/data_types/datatypesoverview.htm

# INTERVAL type

```
2021-12-02 09:21:27,759 WARN sync.util :: Error generating fingerprint for Field 314 'INTERVAL_DAY'
java.time.format.DateTimeParseException: Text '+05 00:00:00.000' could not be parsed, unparsed text found at index 3
        at java.base/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2049)
        at java.base/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1874)
        at metabase.util.date_2.parse$eval13045$parse_with_formatter__13050$fn__13051.invoke(parse.clj:48)
        at metabase.util.date_2.parse$eval13045$parse_with_formatter__13050.invoke(parse.clj:41)
```

# GEOMETRY type

```
2021-12-02 09:24:45,165 INFO sync.util :: STARTING: step 'update-field-values' for exasol Database 6 'vbox1'
2021-12-02 09:24:46,731 ERROR models.field-values :: Error fetching field values
clojure.lang.ExceptionInfo: Error executing query {:sql "-- Metabase\nSELECT \"META\".\"DATA_TYPES\".\"GEO\" AS \"GEO\" FROM \"META\".\"DATA_TYPES\" GROUP BY \"META\".\"DATA_TYPES\".\"GEO\" ORDER BY \"META\".\"DATA_TYPES\".\"GEO\" ASC LIMIT 5000", :params nil, :type :invalid-query}
        at metabase.driver.sql_jdbc.execute$execute_reducible_query$fn__47036.invoke(execute.clj:505)
        at metabase.driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:502)
        at metabase.driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:488)
        at metabase.driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:497)
        at metabase.driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:488)
        at metabase.driver.sql_jdbc$eval61591$fn__61592.invoke(sql_jdbc.clj:54)
...
Caused by: java.sql.SQLException: Feature not supported: GEOMETRY type in GROUP BY (Session: 1717868520831320064)
        at com.exasol.jdbc.ExceptionFactory.createSQLException(ExceptionFactory.java:258)
        at com.exasol.jdbc.EXASQLException.getSQLExceptionIntern(EXASQLException.java:50)
        at com.exasol.jdbc.AbstractEXAStatement.execute(AbstractEXAStatement.java:478)
        at com.exasol.jdbc.EXAStatement.execute(EXAStatement.java:289)
        at com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)
        at metabase.driver.sql_jdbc.execute$eval46940$fn__46941.invoke(execute.clj:369)
...
```

## TIMESTAMP

```
2021-12-02 09:28:19,173 WARN sync.util :: Error generating fingerprint for Field 4,462 'WEB_REC_END_DATE'
clojure.lang.ExceptionInfo: Conversion failed {:path [[#object[java_time.graph.Types 0x63df1fb9 "[java.util.Date]"] #object[java_time.graph.Types 0x3b981537 "[java.time.Instant]"]]], :arguments [#inst "1999-08-15T22:00:00.000-00:00"], :to java.time.Instant}
        at java_time.temporal$instant$fn__10550.invoke(temporal.clj:350)
        at java_time.temporal$instant.invokeStatic(temporal.clj:350)
        at java_time.temporal$instant.invoke(temporal.clj:350)
        at metabase.sync.analyze.fingerprint.fingerprinters$eval49901$fn__49902.invoke(fingerprinters.clj:199)
        at metabase.sync.analyze.fingerprint.fingerprinters$eval49856$fn__49857$G__49847__49862.invoke(fingerprinters.clj:186)
...
Caused by: java.lang.UnsupportedOperationException
        at java.sql/java.sql.Date.toInstant(Date.java:316)
        at java_time.temporal$eval10519$fn__10520.invoke(temporal.clj:332)
...
```

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #6

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ jobs:
 
       - name: Spawn Exasol environment
         run: |
-          ./start-test-env spawn-test-environment --environment-name test --database-port-forward 8563
+          ./start-test-env spawn-test-environment --environment-name test --database-port-forward 8563 --deactivate-database-setup
```
Comment (jakobbraun) [inline .github/workflows/integration-tests.yml:92]: did you intend to commit this?

File: doc/changes/changes_0.2.0.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# metabase-driver 0.2.0, released 2022-01-17
+
+Code name: Fixed issues reading `TIMESTAMP` columns and added support for Exasol specific data types.
```
Comment (jakobbraun) [inline doc/changes/changes_0.2.0.md:3]: Quite long for a release name. Maybe:
```suggestion
Code name: Exasol specific data types and fix for `TIMESTAMP` columns
```

File: doc/changes/changes_0.2.0.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Summary
+
+This release fixes two issues when reading `TIMESTAMP` columns:
+
+* Reading null values caused a `NullPointerException` in the driver.
+* Timestamp values are now read in UTC timezone instead of the local time zone.
+
+The release also adds support for Exasol specific data types `INTERVAL`, `GEOMETRY` and `HASHTYPE`. As Metabase does not support these types, they are returned as strings of Metabase type `:type/*`. There is an issue with scanning values of `GEOMETRY` columns, see [issue #20](https://github.com/exasol/metabase-driver/issues/20) for details.
```
Comment (jakobbraun) [inline doc/changes/changes_0.2.0.md:12]: maybe link the user guide

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (kaklakariada) [reply .github/workflows/integration-tests.yml:92]: Yes. This speeds up the build and we don't need it.
Tests only need a DB, no JDBC driver installed.

See https://github.com/exasol/integration-test-docker-environment/blob/master/README.md#options: 
> Deactivates the setup of the spawned database, this means no data get populated and no JDBC drivers get uploaded.

File: doc/changes/changes_0.2.0.md (hunk 3 above)
Comment (kaklakariada) [reply doc/changes/changes_0.2.0.md]: ✅

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**